node/src/accountant/payment_adjuster/service_fee_adjuster.rs

```diff
@@ -119,27 +122,31 @@ impl ServiceFeeAdjusterReal {
         Vec<UnconfirmedAdjustment>,
     ) {
         let init: (Vec<UnconfirmedAdjustment>, Vec<UnconfirmedAdjustment>) = (vec![], vec![]);
-        let fold_guts = |(mut above_disq_limit, mut below_disq_limit): (Vec<_>, Vec<_>),
-                         current: UnconfirmedAdjustment| {
-            let disqualification_limit = current.disqualification_limit_minor();
-            if current.proposed_adjusted_balance_minor >= disqualification_limit {
-                thriving_competitor_found_diagnostics(&current, disqualification_limit);
-                let mut adjusted = current;
-                adjusted.proposed_adjusted_balance_minor = disqualification_limit;
-                above_disq_limit.push(adjusted)
-            } else {
-                below_disq_limit.push(current)
-            }
-            (above_disq_limit, below_disq_limit)
-        };
+        let fold_guts =
+            |(mut above_or_even_to_disq_limit, mut below_disq_limit): (Vec<_>, Vec<_>),
+             current: UnconfirmedAdjustment| {
+                let disqualification_limit = current.disqualification_limit_minor();
+                if current.proposed_adjusted_balance_minor >= disqualification_limit {
+                    diagnostics_for_accounts_above_disqualification_limit(
+                        &current,
+                        disqualification_limit,
+                    );
+                    let mut adjusted = current;
+                    adjusted.proposed_adjusted_balance_minor = disqualification_limit;
+                    above_or_even_to_disq_limit.push(adjusted)
+                } else {
+                    below_disq_limit.push(current)
+                }
+                (above_or_even_to_disq_limit, below_disq_limit)
+            };
```

Comment:
I think this could be further refactored like this:

```rust
let fold_guts = |(mut above_limit, mut below_limit): (Vec<_>, Vec<_>), current: UnconfirmedAdjustment| {
    let disqualification_limit = current.disqualification_limit_minor();
    
    if current.proposed_adjusted_balance_minor >= disqualification_limit {
        diagnostics_for_accounts_above_disqualification_limit(&current, disqualification_limit);
        
        let adjusted = UnconfirmedAdjustment {
            proposed_adjusted_balance_minor: disqualification_limit,
            ..current
        };
        above_limit.push(adjusted);
    } else {
        below_limit.push(current);
    }
    
    (above_limit, below_limit)
};
```